packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -57,7 +57,10 @@ export default util.createRule<Options, MessageIds>({
       if (type.isNumberLiteral()) {
         return 'number';
       }
-      if (type.isStringLiteral()) {
+      if (
+        type.isStringLiteral() ||
+        util.isTypeFlagSet(type, ts.TypeFlags.TemplateLiteral)
+      ) {
         return 'string';
       }
```

Comment:
should we make these two checks instead something like this?
that'd cover both cases for free.

```ts
if (util.isTypeFlagSet(type, ts.TypeFlags.NumberLike)) {
  return 'number';
}

if (util.isTypeFlagSet(type.flags, ts.TypeFlags.StringLike)) {
  return 'string';
}
```

Reply:
Alas, `string | undefined` doesn't trigger for `StringLike`. I don't see a way to get just one check.

```
      ✕ 
let foo: string | undefined;
foo += 'some data';
       (6 ms)
```